src/components/content/policies/PolicyContentDisplay.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import { Alert, UploadFile } from 'antd';
+import React from 'react';
+import DisplayPolicyData from './DisplayPolicyData';
+
+function PolicyContentDisplay(policyContent: string, file: UploadFile): React.JSX.Element {
+    const policyData = DisplayPolicyData({ policyContent: policyContent });
+
+    if (typeof policyData === 'string') {
+        file.status = 'error';
+        return (
+            <Alert
+                type={'error'}
+                showIcon={true}
+                message={`Policy content data in the uploaded file not Valid. Error while parsing - ${policyData}`}
+            />
+        );
+    } else {
+        file.status = 'done';
```

Comment:
if i understand it right, file.status should be done only after the add or delete is successful.